subsys/bluetooth/mesh/Kconfig

```diff
@@ -434,6 +443,25 @@ config BT_MESH_SHELL
 	  Activate shell module that provides Bluetooth Mesh commands to
 	  the console.
 
+config BT_MESH_NODES
+	bool "Support for saving node information"
```

Comment:
Should this depend on `BT_MESH_PROVISIONER`? Is this about runtime or flash storage? If it's the latter I suppose a `BT_SETTINGS` dependency is also needed. Update: I see you're doing the reverse, i.e. selecting this option from `BT_MESH_PROVISIONER`. See my other comment about that.

I understand the need to come up with a short name for this feature, and "NODES" is not a bad attempt. I find it slightly confusing though since the local device is typically a node itself. To me at least, this all sounds like a provisioning database, so would perhaps `BT_MESH_PROV_DB` sound intuitive?

Reply:
> I understand the need to come up with a short name for this feature, and "NODES" is not a bad attempt. I find it slightly confusing though since the local device is typically a node itself. To me at least, this all sounds like a provisioning database, so would perhaps `BT_MESH_PROV_DB` sound intuitive?

I like BT_MESH_PROV_DB. Do you think the name `node` or `nodes` in the code is ok? I can really think of a good name to distinguish between the local node and other nodes.

Reply:
`node`/`nodes` in the code is fine with me, however if there's no good justification for keeping a separate `BT_MESH_PROV_DB` Kconfig option around (i.e. a reason why you'd sometimes want one but not the other) I'd just stick to a single `BT_MESH_PROVISIONER` option.

Reply:
`BT_MESH_NODES` is removed but `BT_MESH_NODE_COUNT` is still left. Do you think this is ok or should it be maybe `BT_MESH_PROV_DB_SIZE` or something?